### PR TITLE
[DeviceAPI] revisit Executor to make this use on Device API

### PR DESCRIPTION
**Pull request by struss**
This commit will revisit `Executor` for some reason

about `name()`: if we add None Backend Executor, we need to add Executor name on this.
                          otherwise, follow Backend Name.
about `require()`: to use on `satisfied()` on Device API.

ONE-vscode-DCO-1.0-Signed-off-by: struss <rrstrous@nate.com>

### CODE REVIEW

**Review by YongseopKim — APPROVED** [review]
LGTM

File: src/Backend/Executor.ts [hunk 1]
```diff
@@ -26,10 +30,17 @@ interface Executor {
...
   runInference(_modelPath: string, _options?: string[]): Command;
+
+  // return deviceSpec that required for this Executor to run
+  require(): DeviceSpec;
 }
 
 // General excutor uses onecc so default jobs can be used
 class ExecutorBase implements Executor {
+  name(): string {
+    throw new Error('Name are not determinded.');
```
Comment (hyunsik-yoon) [inline src/Backend/Executor.ts:41]: ??

```suggestion
    throw new Error('Name is not determined.');
```

File: src/Backend/Executor.ts [hunk 2]
```diff
@@ -26,10 +30,17 @@ interface Executor {
 
   // TODO: use cfg path to run onecc after one-infer landed
   runInference(_modelPath: string, _options?: string[]): Command;
+
+  // return deviceSpec that required for this Executor to run
```
Comment (hyunsik-yoon) [inline src/Backend/Executor.ts:34]:
```suggestion
  // return deviceSpec required for this Executor to run
```

Comment (hyunsik-yoon) [reply src/Backend/Executor.ts:34]: or 

```suggestion
  // return deviceSpec that is required for this Executor to run
```

File: src/Backend/Executor.ts [hunk 3]
```diff
@@ -55,7 +55,7 @@ class ExecutorBase implements Executor {
   }
 
   require(): DeviceSpec {
-    throw new Error('Invalid require Spec call.');
+    throw new Error('Invalid DeviceSpec call.');
```
Comment (hyunsik-yoon) [inline src/Backend/Executor.ts:58]: `DeviceSpec` is a return type and I am not sure whether `'Invalid DeviceSpec call.'` is appropriate.

For example, if we have a method like the following, 
`'invalid int call` sounds not appropriate.

```
int add(a, b) {  // return int type
   if (some error situation)
      throw new Error('invalid int call')
}
```

Comment (struss) [reply src/Backend/Executor.ts:58]: um.... I will fix it.

**Review by hyunsik-yoon — APPROVED** [review]

**Review by YongseopKim — APPROVED** [review]

**Pull request merged**